### ISSUE TITLE
Proposed RA Balance Changes 2019

### ISSUE THREAD

**Issue by Punsho** [OpenRA/OpenRA#16418]
# SFNO (Stealth & Faction & Naval Overhaul)

I've been working on RA balance for about a month. Here I streamline stealth, rework England / France / Germany faction specific traits, overhaul Naval balance and address new issues caused by the previous update. It also contains planned features such as Hind to Soviet

So far it was well received by the competitive community. If anyone has concerns or questions don't hesitate to contact me. **Here's the [link](https://docs.google.com/document/d/1_IlVGSbax5V2T9x76MsNDjkTGvyM9FJbChwux93o8ME/edit?usp=sharing) to the change log with explanations and test maps.** 

# These are the raw changes (outdated)
## Streamlining stealth:
   - Removed detection from Radar Domes
   - Removed detection from Tanya
   - Removed detection from Phase Transport
   - Removed detection from Mammoth Tanks
   - Removed detection from Mobile Radar Jammer 
   - Removed detection from Spy Plane
   - Minelayer is given cloak detection
   - Mines are made detectable by all cloak detectors
   - Thieves can no longer be detected by infantry
   - Sonar Pulse has been moved to Naval Yard
   - Spy Plane now detects Submarines

### General
- Veterancy
   - Rank count from 4 to 3
   - Ranks are visible to opponents
   - Progression changed from 5% per rank to 10% per rank
   - Movement speed buffs removed from lower ranks, only 30% increase on elite
   - Removed accuracy improvements
   - Faster health regeneration on elite rank
   - Removed healing speed limit
- Sandbags / Wire fence: Value from 50 to 30
- All armaments that don't have a visible projectile except for Mobile Flak and Sea Scorpion are made instant hit
- High tech units no longer build faster then other units: Tesla & Chorno & Mammoth tank, Phase Transport, MRJ, MGG, Mig, Longbow, Cruiser, Missile Submarine
#### Units
- Spy: Can steal money from Refineries
 - new Black Hawk helicopter: same stats as current Hind, name tbd
 - new Sniper: 
    - Valid Targets: Infantry only
    - Damage: kills all infantry in one attack
    - Value: 700
    - Health: 10000 
    - Cloak:
      - CloakDelay: 2 seconds
    -  Prerequisite: England, Allied Tech Center
    - Misc: 
      - Fire rate is twice as slow as shock troopers
      - Can switch between 2 states
      - Cloaked and only uncloaked when shooting, 3c0 vision and weapon range
      - Uncloaked but has 7 vision and weapon range (Tanya has 7 weapon range and 6 vision) Uncovers his face (visual)
  -  Mobile Gap Generator: 
     - GrantConditionOnDeploy: Toggle shroud generation
  - British Spy: removed
#### SupportPowers
 - Advanced Chronoshift to all Allied factions
 - new Tank Drop:
   - 3 light tanks every 5 minutes
   - Prerequisite: Germany, Radar Dome

#### Units
 - Hind: 
   - Production: airfield
   - Cargo: 4 infantry
   - Armament: Inherits ChainGun with these modifications
      - ReloadDelay: from 10 to 25
      - Damage: from 2500 to 1000
   - Value: 1000
   - RevealsShroud: 9c0
 - Mammoth Tank:
   - Armement: Tusk Missile range from 8c0 to 6c512
 - Thief: 
    - Prerequisite: Radar Dome
    - Health: from 5000 to 4500
    - Health does not affect Cloak
  - Dog:
    - Vision: from 5c512 to 4c0
    - Health: from 1800 to 7000
    - Slow regeneration

#### Units:
 - Gunboat:
   - Reload delay: from 60 to 20
   - Projectile speed: from 426 to 724
   - Attack range: from 5c512 to 4c512
   - Cost: from 500 to 600
   - Submarine detection: from 4c0 to 3c0
 - Depth Charge:
   - Damage: from 8000 to 3000
   - Damage vs light armour: from 75% to 100%
 - Destroyer:
   - Prerequisite: Naval Yard
   - Submarine detection: Removed 
   - Depth Charges: Removed
   - Health: from 40000 to 35000
 - Black Hawk:
    - new Armament: Depth Charge
      - Damage spread: from 128 to 502
#### Structures:
 - Turret :
   - Weapon range: from 7c0 to 7c512
 - Naval Yard: 
    - Cost: from 1000 to 650

#### Units:
 - Submarine
    - Can submerge at player’s will
      - When submerged: Low vision of 5c0
      - When on the surface: High vision of 9c0
    - Submarine detection range: from 4c0 to 5c0
    - Torpedo 
      - HorizontalRateOfTurn: from 1 to 2
      - Damage: from 18000 to 10000
      - Damage vs armour types:
        - Wood: from 75% to 50%
        - Light: from 75% to 50%
        - Heavy: from 100% to 85%
        - Concrete: from 500% to 850%
    - Initial stance: from hold fire to defend
  - new Sea Scorpion:
    - Health: 20000
    - Cost 600
    - Movement Speed: 116
    - Turn speed: 10
    - Vision: 6c0
    - Facing tolerance: 2
    - Armament: inherits Flak weapon with these modifications:
      - Armour modifiers:
        - Wood: from 10% to 50%
        - Light: from 60% to 50%
        - Heavy: from 10% to 25%
        - Concrete: from 30% to 40%
      - Anti-ground weapon range 7c512
    - Armour type: heavy
  - Missile Submarine:
    - Vision: from 8c0 to 5c0
    - Anti-air missiles: removed 
 - Hind:
    - new Armament: Depth Charge
      - Damage spread: from 128 to 502
#### Structures:
 - Sub Pen:
    - Cost: from from 850 to 650

This can close #16424 #16038 #15100 #14100
Current PRs: #16514 #16472

**Comment by matjaeck** [comment]
Thanks for updating the comment, this looks much better now. Some of the changes will be very controversial and the overall amount of changes is too much for one balance PR. I would suggest to exclude the very very controversial changes like the sniper, tank drop, intel drop and minimap changes for now. These are not balance changes but new features and it is very unlikely that the attempt to sneak them in with other (controversial) balance changes will be successful. But even after this there remain way too many changes - I don't know how this should ever get through the review process if you don't massively cut down the changes.

IMO focus on helicopters, thief and mammoth and exclude everything else.

Please give the other people here a chance to understand what you are doing and the opportunity to participate by opening issues for the problems you are trying to solve. You need their input and their support.

**Comment by Punsho** [comment]
@matjaeck I have considered that other people could be working on it and I've been following their progress. For example the Black Hawk sprite I use was made to replace hind with pchote reviewing it on Discord. I'm thinking about a replacement for the tank drop. The current community ideas are tank chronoin (some tanks would chrono in you could control them and chrono out when time runs out) and chrono vortex. Germany & France & England specific traits need to be reworked and it can't be done without adding new units / support powers powers. Naval needs a long list of changes to even start being balanced. I Include some small changes here too such as Submarines being able to kill walls. Could make separate pr's for them to make this list shorter if necessary

For now I will continue testing this balance and decreasing it's size only after a review process or when I find better solutions to problems

**Comment by matjaeck** [comment]
The things I noted have all been communicated to you earlier by multiple people, you seem to be resistant to their advice though. I have big doubts, after reading your comment here and your latest forum posts (in which you also seem to ignore other people's suggestion and exclusively focus on your vision while considering your opinion in line with a "perfect" state of things) that you are ready to participate in a big open source project like OpenRA is. You are of course free to do whatever you want, but it will not happen that you damage the spirit of this project with your involvement.

**Comment by Punsho** [comment]
It might seem this way because most of the discussions and support I get are from discord, private messages, game lobbies

**Comment by Punsho** [comment]
Oh and also I made this issue when most concerns and issues people had had been addressed and accounted for

**Comment by matjaeck** [comment]
On a sidenote, we need some balance changes merged for next release. My main point is to decrease the mammoth tusk missile range like suggested above. This change has been discussed on the forum and the community overall agreed with it I think. If I get the feeling that the "normal" balance PR will be impossible to review, I'll file this change myself.

**Comment by MustaphaTR** [comment]
I'm not someone who plays too often, so i'm not sure how much word i can have on balance, but i wanna say my opinion on stuff here.

> Streamlining stealth:

I'm neutral on most of those, except imo Radar should keep the stealth detection.

> Radar on ConYard:

I don't like this idea in an design perspective rather than balance. As i said on discord i would be complately OK with a Free Radar checkbox (disabled by default). Or we implement something like in Dune 2, which you can see your buildings on the radar even without Outpost.

> Sandbag/Wire Fence Cost:

I'm not sure how much it is gonna help but i'm ok with that change.

> Mine damage and instant hit projectiles.

I see those as bug fixes that will have balance effect than actual balance changes. Agree with the changes.

> Spy stealing from refinery:

I'm not 100% sure, but i'm positive on the change.

> BlackHawk/Hind changes.

That was pretty much what i wanted to be done with the Hind. Definetly agree. But could be done in a seperate PR than the actual balance changes.

> Sniper, Adv. Chronoshift, Tank Drop, Intel Drop, Soviet Support Powers being more to Radar.

Neutral on the topic. If added, should be handled in its own PR imo.

> MGG Deploy ability.

Should help to have this.

> Mammoth Tank Missile

Positive on the change.

> Thief/Dog

Neutral on the changes. I'm not sure about the current state of the units.

> Naval Changes:

My overall state on the Soviet Navy have always been that they need a new Siege Submarine and Missile Sub should be repurpsed to be a better AA. But your sounds better than the current state too.

> Subpen/Shipyard Cost:

Definetly should be changed to match.

> Turret Range:

Sounds good to me.

**Comment by Smittytron** [comment]
The group of changes above would be a major overhaul that doesn't mesh with the direction we've taken balance since Soscared's first balance patch. New units and support powers especially. I won't be able to include most of what's up there in my eventual balance PR.

**Comment by Punsho** [comment]
The core gameplay practically doesn't change, naval is an exception ofc. I already have maps with all of these changes done so there won't technical problems. We can scrap the new support powers but what to put in their place? England is currently the weakest faction and adding in the sniper could change it. I can revert globalisation of mgg thus removing the Intel drop. Then we come to German Chronosphere. The normal chrono needs a buff and a good way to do it is by globalising the German one. It leaves an empty gap and currently there are currently 3 ideas I mentioned that can fill it

**Comment by Punsho** [comment]
The core gameplay practically doesn't change, naval is an exception ofc. I already have maps with all of these changes done so there won't technical problems. We can scrap the new support powers but what to put in their place? England is currently the weakest faction and adding in the sniper could change it. I can revert globalisation of mgg thus removing the need for Intel drop. Then we come to German Chronosphere. The normal chrono needs a buff and a good way to do it is by globalising the German one. It leaves an empty gap and currently there are 3 ideas I mentioned that can fill it

**Comment by ZxGanon** [comment]
I´m agreeing with Mustapha on most of his quotes.

Punsho is gonna tackle lots of issues/long standing ignored design flaws that RA currently has like Naval in general, Stealth Detection inconsistency and most important the overall Design of the Alliessubfactions that has to be addressed.

We will have to play/gather lots of games to see how his ideas take effect on RA1.

Punsho is a very active member in the community and does listen to the feedback of people so discarding him now would be a huge loss for the future of OpenRA.

**Comment by matjaeck** [comment]
I just fail to understand what makes it so hard to file an issue titled "Stealth is inconsistent" or "Rework allied subfactions" with a reasonable issue summary.

**Comment by Punsho** [comment]
@matjaeck I hope we do. When I wrote my changes I tried to explain them as simple and straight forward as possible. Connecting all dots and not leaving much for interpretation. After that I questioned tens of people if they found anything in my balance unclear and what and why they disliked. So far you're the first one that didn't understand my intentions so do you have tips on how I should improve my wording?

**Comment by matjaeck** [comment]
@Punsho I will probably focus on other topics that I find more interesting and easier to support so if everyone else is fine with your changes, I'm not going to be the one guy blocking progress, but also not the one helping to get them merged. I have trust in Smitty that he'll handle this.

**Comment by Th1sIsSoSad** [comment]
🙌 

Over the years the "balance" changes have flattened differences between OpenRA units.

Actually all 1v1 games look almost the same, we always see the same army compositions, the same timings, and they are all capped to tier 2.
It's basically arties/hinds + tier 1 infantry vs heavies/v2 + tier 1 infantry. Sometimes we see an engineer or a thief somewhere.

We also barely see tier 3 units because tier 3 doesn't add much value.
Having a very strong (and costly) tier 3 unit on the battlefield would make players build it (maybe rush it) and protect it (a bit like "heroes" in some other games).

Please make tier 3 worth it again, for every faction.

Please bring back the mammoth tank we had in 2013-2014 ! 😭

**Comment by GraionDilach** [comment]
The Blackhawk cameo needs tweaking, it's text bar needs to a pixel taller to fit in with all other cameos.

**Comment by Punsho** [comment]
Good catch, it's name is still not final

**Comment by matjaeck** [comment]
After more discussions around the proposed changes here I'd like to sum up my thoughts on a potential rework of general and faction specific support / super powers:

As pointed out in many places allies are not only in a weaker spot balance wise at the moment, but also lack in terms of game experience. Their faction specific build options / support / super powers do not reach the fun factor and the effectiveness of the Russian or Ukrainian arsenal.

Since adding new units is always very difficult and problematic and porting features from later games likewise is, utilizing existing concepts and artwork makes more sense. 

The discussion on Discord has resulted in the proposal to implement an ability to call in temporary chrono-reinforcements consisting of light and or medium tanks. Based on the proposed changes here and the discussions on discord I have these considerations:

This ability could be made available as a faction specific ability (for example Germany) while replacing their advanced chronosphere. The chronosphere would it turn be normalized across all allied factions and buffed to be similar to the old advanced chronosphere, since the current regular chronosphere is broadly considered useless compared to the iron curtain.

To fix the situation of England, additional invasive changes appear to be required. I don't really like that we take the MGG out of the standard allied arsenal but it is the only remaining vehicle that we can give them. Both ideas to toggle shroud generation or to change how enemy units respond to shroud have their problems / will need further discussion so I suggest skipping this for now. Regarding England's second trait, the recently introduced directional support powers could be utilized in form of a spy paradrop with tbd details (disguised / count). This would be compatible with the suggested ability of resetting super weapons timers as these buildings have no bibs to drop off and can be walled off easily.

France seems to be already interesting so can stay as it is.

**Comment by matjaeck** [comment]
Possibly closed by #16803.

**Comment by Punsho** [comment]
sure
